_plugins/content-modifier.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+require "jekyll/hooks"
+
+##
+# This singleton modifies the content at build time. Its primary use is to mitigate security risks automatically.
+#
+# The plugin does not apply changes when the Serve command is used but `JEKYLL_ALLOW_CONTENT_MODIFIER`, set on the
+# environment, will override the behavior.
+# Usage: `JEKYLL_ALLOW_CONTENT_MODIFIER= bundle exec jekyll serve --trace`
+
+module Jekyll::ContentModifier
+
+  ##
+  # Pattern to identify documents that should be excluded based on their URL
+  @excluded_paths = /\.(css|js|json|map|xml|txt|yml|svg|)$/i.freeze
+
+  ##
+  # Defines the priority of the plugin
+  # The hooks are registered with a very high priority to make sure the changes are in before other hooks are run
+  def self.priority
+    90
+  end
+
+  ##
+  # Initializes the singleton by recording the site
+  def self.init(site)
+    @site = site
+
+    # Avoid initializing if serving and not forced to run
+    if site.config["serving"] and (!ENV.key?('JEKYLL_ALLOW_CONTENT_MODIFIER') or ENV['JEKYLL_ALLOW_CONTENT_MODIFIER'] == "false")
```

Comment:
Can we drop a line or two into README about how to turn it on and off (and what it does)?

Reply:
Done.